### ISSUE TITLE
Can't use EOF heredoc

### ISSUE THREAD

**Issue by chrisokamoto** [bbatsov/rubocop#5403]
In the docs it says: 

> This cop checks that your heredocs are using meaningful delimiters. By default it disallows END and EO*, and can be configured through blacklisting additional delimiters.

But in the AllowedAcronyms' table, it says that EOF is allowed, but in fact, it isn't. When I try to use EOF, it says: 

> Naming/HeredocDelimiterNaming: Use meaningful heredoc delimiters. EOF

But I am not sure if I misunderstood it.
```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.4.1 x86_64-linux)
```

**Comment by mcfisch** [comment]
The rule doesn't allow anything starting with 'EO' or containing 'END'. The rule isn't specific enough, and therefor actually doesn't really allow ANY meaningful delimiters. Changing it to this fixes it:

```Naming/HeredocDelimiterNaming:
  Blacklist:
    - ^END$
    - !ruby/regexp '/^EO[A-Z]{1}$/'
```

**Comment by mcfisch** [comment]
This would block all 3-char delimiters starting with `EO` as well as `END`, but still allow `EOF`.

**Comment by mcfisch** [comment]
Created #5712 to fix this. Tests passed, also works fine in my local `~/.rubocop.yml`.

**Comment by Drenmi** [comment]
`EOF` was intentionally disallowed as a non-meaningful delimiter of this cop. The acronym list is for an unrelated cop. 🙂 

If you want to use `EOF` in your project, the right way is to use the local configuration option. 😉

**Comment by mcfisch** [comment]
Ok, so it isn't related to the acronyms list, they just reside next to each other because they belong to the same category. Ok, I'll change the code accordingly.

However, the current implementation doesn't allow to have any delimiter starting or even **containing** `EO[A-Z]` or `END`, and I can't imagine this is on puropose, is it?

**Comment by Drenmi** [comment]
The reason is it's an "empty" word, which doesn't help provide context for what the contents of the string is. Compare:

```ruby
foo = <<-EOF
  ...
EOF
```

to:

```ruby
foo = <<-SQL
  ...
SQL
```

or:

```
foo = <<-ERROR
  ...
ERROR
```

In both the latter cases, you get an idea of what this string is, even without looking at the content or the variable name. In the case of `SQL`, most editors will also apply code highlighting to the string content.

**Comment by mcfisch** [comment]
The problem with the current definition is, that it even faults something like this just because it _contains_ the banned pattern:

```
foo = <<-ENDPOVERTY
  ...
ENDPOVERTY
```
as well as 

```
foo = <<-ISLEOFMAN
  ...
ISLEOFMAN
```

**Comment by mcfisch** [comment]
I've changed the pattern to include `EOF` as well.

**Comment by Drenmi** [comment]
> The problem with the current definition is, that it even faults something like this just because it contains the banned pattern.

Oh. Yes. This is definitely not the intention. Good catch! 🙂

**Comment by mcfisch** [comment]
Thx for the merge, couldn't get back to it earlier.
